### ISSUE TITLE
DAS without the anaconda channel

### ISSUE THREAD

**Issue by postpop** [janclemenslab/das#78]
Many institutions are blocking the anaconda channel because of licensing issues, so we need builds that work without anaconda. This is mainly an issue for windows because we rely on anaconda to provide the tensorflow-gpu package.

The new release 0.38.4 was built without anaconda in all supported platforms (linux, macos (arm only), windows).

You can give it a try with:
`conda create -n das python=3.10 das=0.32.4 -c conda-forge -c ncb`

Linux and Mac builds should also work with `python=3.11`.

Please test and report any issues with the no-anaconda builds here.

May also address #77.

**Comment by avakiai** [comment]
Just commenting that this worked very well for me. Just a note: don't be surprised if the install takes much longer than previous installations of das...

**Comment by postpop** [comment]
Thanks! Glad it works.

And thanks for the note on the slowness: It didn't take longer on my machine. What version of conda are you on? The newer versions (23.10+) use the faster mamba solver by default (https://conda.org/blog/2023-11-06-conda-23-10-0-release/). 

So either update conda (`conda update conda -n base`) or use mamba instead of conda for the installation. We will update the docs with this information once the new builds have been tested properly.

**Comment by hummuscience** [comment]
I am getting errors that the ncb channel does not include files for osx-arm64

**Comment by postpop** [comment]
Hi, please use the installation instruction from the github readme or the documentation.
This is the correct command:
```
conda create python=3.11 das=0.32.4 -c conda-forge -c ncb -c nvidia -c apple -n das -y
```
Just tested it on my M1 mac and it works. Please re-open if you run into issues with this.

**Comment by postpop** [comment]
Ah, I see. I don't think your institute needs to do that. To avoid licensing issues, all they need to do is block the anaconda channel (https://conda.anaconda.org/anaconda), not the user channels. The above command works with an MPI and our institute - so I guess they block channels more selectively. 

Don't think there is much I can do about this atm. I could share an environment file, but even that requires access to the ncb channel.

In the future, we might try to get DAS onto the conda-forge channel, but this is a lengthy process and would make deployment more complicated.

**Comment by hummuscience** [comment]
Thanks for checking. I will talk with IT and see if I can find a solution.

**Comment by postpop** [comment]
Thanks! Would be good for me and other's to know! 

In the meantime, I will explore how to get DAS onto conda-forge.

**Comment by avakiai** [comment]
@hummuscience Did you update your  ~/.condarc file to remove "defaults" from the channels? (IT sent an email about this in September). That should allow it to get around the firewall and only query conda-forge, if I've understood things correctly.

**Comment by postpop** [comment]
Thanks for the note. Yes, "defaults" were removed from [condarc](https://github.com/janclemenslab/das/blob/master/condarc.yml).

The problem is that DAS needs to query the ncb channel, not just the conda-forge channel. For some reason, that one also seems to be blocked.

**Comment by hummuscience** [comment]
@avakiai yes. Defaults is removed from condarc. Did you manage to install DAS on your machine?
(@postpop we work in the same institute)

**Comment by bjudkewitz** [comment]
Thanks to everyone in this thread – it helped me get GPU support working. I initially tried the installation steps from the docs (https://janclemenslab.org/das/installation.html) but couldn’t get the GPU to work.

For reference, what worked for me in the end on Linux was:
`mamba create -n das ncb::das conda-forge::tensorflow-gpu`

In case it’s useful: a small note in the installation page (or even just a link from the docs to this issue) might help future users who run into the same situation. Totally understand maintainers’ time is limited – linking here alone would have saved me a lot of trial and error.

**Comment by postpop** [comment]
Deploying DAS with TensorFlow is a real pain and leads to hard-to-reproduce installation errors. On some of the machines we tested it works out of the box; on others it doesn’t. We have a PyTorch-backend version of DAS almost ready, which should solve these issues, make DAS more future-proof, and enable new features.

Until this is ready for release, I’ll add a link to these issues in the docs.
